kubernetes/kubernetes.go

```diff
@@ -186,14 +185,22 @@ func (ka KubeAgentConfig) collectMetrics(
 	// get raw Heapster metric sample
 	hme, err := rawClient.GetRawEndPoint("heapster-metrics-export", metricSampleDir, config.HeapsterURL)
 	if err != nil {
-		return fmt.Errorf("error retrieving Raw Heapster metrics: %s", err)
+		return fmt.Errorf("unable to retrieve raw heapster metrics: %s", err)
 	}
 
 	defer util.SafeClose(hme.Close, &rerr)
 
-	err = handleBaselineHeapsterMetrics(msd, baselineMetricSample, hme.Name())
+	baselineMetricSample, err := filepath.Glob(path.Dir(config.msExportDirectory.Name()) + "/baseline-metrics-export*")
+
 	if err != nil {
-		return err
+		log.Printf("Error encountered reading export directory: %v", err)
+	}
+
+	if len(baselineMetricSample[0]) >= 1 {
```

Comment:
could this nil point if the call to file.Glob errored out? we check for that, but only log the error (line 196), and don't return.